src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -20,42 +20,39 @@
-                 (request-handler request))
-     (comidi/GET ["/file_metadatas/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/file_metadata/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/file_bucket_file/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/PUT ["/file_bucket_file/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/HEAD ["/file_bucket_file/" [#".*" :rest]] request
-                  (request-handler request))
-
-     (comidi/GET ["/catalog/" [#".*" :rest]] request
-                 (request-handler-with-code-id request))
-     (comidi/POST ["/catalog/" [#".*" :rest]] request
-                 (request-handler-with-code-id request))
-     (comidi/PUT ["/report/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/resource_type/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/resource_types/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET ["/environment/" [#".*" :rest]] request
-                 (request-handler request))
-     (comidi/GET "/environments" request
-                 (request-handler request))
-     (comidi/GET ["/status/" [#".*" :rest]] request
-                 (request-handler request)))))
+  [request-handler]
+  (comidi/routes
+   (comidi/GET ["/node/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_content/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_metadatas/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_metadata/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/file_bucket_file/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/PUT ["/file_bucket_file/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/HEAD ["/file_bucket_file/" [#".*" :rest]] request
+                (request-handler request))
+
+   (comidi/GET ["/catalog/" [#".*" :rest]] request
+               (request-handler (assoc request :include-code-id? true)))
+   (comidi/POST ["/catalog/" [#".*" :rest]] request
+                (request-handler (assoc request :include-code-id? true)))
+   (comidi/PUT ["/report/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/resource_type/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/resource_types/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET ["/environment/" [#".*" :rest]] request
+               (request-handler request))
+   (comidi/GET "/environments" request
+               (request-handler request))
+   (comidi/GET ["/status/" [#".*" :rest]] request
+               (request-handler request))))
```

Comment:
This is *really* hard to review... can we please try to be more diligent about stashing and doing whitespace changes in separate commits?